apis/Google.Cloud.Firestore/Google.Cloud.Firestore.Tests/QueryTest.CreateDocumentSnapshotComparer.cs

```diff
@@ -0,0 +1,209 @@
+                CreateSnapshot("doc3", 3, 1),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc1", 10, 5)
+            );
+        }
+
+        [Fact]
+        public void OrderByValue_AllUnique()
+        {
+            var query = s_db.Collection("col").OrderBy("value1");
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc3", 3, 1),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc1", 10, 5),
+                CreateSnapshot("doc4", 15, 15)
+            );
+        }
+
+        [Fact]
+        public void OrderByValueDescending_AllUnique()
+        {
+            var query = s_db.Collection("col").OrderByDescending("value1");
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc4", 15, 15),
+                CreateSnapshot("doc1", 10, 5),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc3", 3, 1)
+            );
+        }
+
+        [Fact]
+        public void OrderByValue_DocumentIdForTieBreak()
+        {
+            var query = s_db.Collection("col").OrderBy("value1");
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc3", 3, 1),
+                CreateSnapshot("doc1", 8, 5),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc4", 15, 15)
+            );
+        }
+
+        [Fact]
+        public void OrderByValueDescending_DocumentIdForTieBreak()
+        {
+            var query = s_db.Collection("col").OrderByDescending("value1");
+            // Last ordering was descending, so doc2 comes before doc1.
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc4", 15, 15),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc1", 8, 5),
+                CreateSnapshot("doc3", 3, 1)
+            );
+        }
+
+        [Fact]
+        public void OrderByTwoValues()
+        {
+            var query = s_db.Collection("col").OrderBy("value1").OrderBy("value2");
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc4", 3, 5),
+                CreateSnapshot("doc1", 8, 5),
+                CreateSnapshot("doc3", 8, 5),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc5", 15, 15)
+            );
+        }
+
+        [Fact]
+        public void OrderByTwoValues_SecondIsDescending()
+        {
+            var query = s_db.Collection("col").OrderBy("value1").OrderByDescending("value2");
+            AssertDocsInOrder(query,
+                CreateSnapshot("doc4", 3, 5),
+                CreateSnapshot("doc2", 8, 9),
+                CreateSnapshot("doc3", 8, 5),
+                CreateSnapshot("doc1", 8, 5),
+                CreateSnapshot("doc5", 15, 15)
+            );
+        }
+
+        private static void AssertDocsInOrder(Query query, params DocumentSnapshot[] docs)
+        {
+            var rng = new Random();
```

Comment:
Might be better to use a fixed seed; so the test is reproducible?

Reply:
I wondered about that. It would be reproducible within a single runtime, at least. Sure, will do.